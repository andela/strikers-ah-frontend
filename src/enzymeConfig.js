import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
