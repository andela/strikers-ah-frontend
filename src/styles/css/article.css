@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
.spiner {
  position: fixed;
  z-index: 1031;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #43c0f1; }

.comments-container {
  padding: 1%;
  margin-top: 1%;
  border-radius: 1%; }
  .comments-container .comment-count {
    font-weight: bold;
    margin-left: 4%; }

html {
  height: 100%;
  margin: 0; }

body {
  font-family: "Arial", "Helvetica", "sans-serif";
  font-size: 15px;
  margin: 0;
  padding: 0;
  /* background-color: darkgray; */
  min-height: 100%; }

.container {
  width: 80%;
  margin: auto;
  overflow: hidden; }
  @media (max-width: 768px) {
    .container {
      width: 100%; } }

header {
  background-color: #ffffff;
  padding-top: 10px;
  min-height: 70px;
  border-bottom: #e5e5e5 3px solid; }

.brand {
  margin-top: 10px;
  float: left; }

header .icons ul {
  list-style-type: none;
  margin: 0;
  padding: 0; }

header .icons ul li i {
  color: grey; }

header .icons {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: flex-end; }
  @media (max-width: 768px) {
    header .icons {
      display: none; } }

.brand .brandName {
  text-decoration: none; }

.brand .brandName {
  font-size: 25px;
  font-family: Georgia, 'Times New Roman', Times, serif;
  color: #00aef9; }

.content {
  background-color: #e5e5e5;
  align-content: center;
  margin-right: 100px;
  margin-left: 100px; }
  @media (max-width: 768px) {
    .content {
      margin: auto;
      margin-right: 5%;
      margin-left: 5%; } }

.content .editor-tools .container .bar-tool #align-bartool {
  text-align: center; }

.content .editor-tools .container .bar-tool {
  margin-top: 10px;
  margin-bottom: 10px;
  background-color: #ffffff;
  border-radius: 8px;
  box-shadow: 0px 4px 4px #c4c4c4;
  padding: 10px;
  display: inline-block;
  /* position: absolute; */ }
  @media (max-width: 768px) {
    .content .editor-tools .container .bar-tool {
      margin: auto;
      padding: 5px; } }

.content .editor-tools .container .bar-tool {
  margin-bottom: 100px;
  margin-left: 10%;
  margin-right: 10%; }

input[type='color'] {
  border: none; }

.content .editor-tools .container .bar-tool #border {
  border: 1px solid #c4c4c4;
  transform: rotate(90deg); }

.content .editor-tools .container .bar-tool > button {
  margin-right: 5px;
  margin-left: 5px;
  border: no; }

.content .editor-tools .container .bar-tool > button:hover {
  background-color: #dfe6e9;
  transition: all 0.3s linear 0s;
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  cursor: pointer; }

.content .editor-tools .container .bar-tool #highlight {
  margin-right: 5px;
  margin-left: 5px; }

.content .text-area {
  width: 100%; }

[contentEditable='true']:empty:not(:focus):before {
  content: attr(data-text);
  color: gray; }

.content .text-area .form #title {
  background-color: #ffffff;
  margin-bottom: 5px;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%; }

.content .text-area .form #body {
  margin-bottom: 10px;
  background-color: #ffffff;
  align-content: center;
  margin-top: 0%;
  width: 100%;
  height: auto;
  border: none;
  outline: none;
  padding: 30px;
  font-size: 30px;
  font-weight: 400px;
  font-family: "Arial", "Helvetica", "sans-serif";
  border-radius: 3px;
  line-height: 1.5em;
  color: #1e272e;
  margin-right: 10%;
  min-height: 800px; }

.content .text-area .form #publish {
  cursor: pointer;
  border: none;
  background: #808e9b;
  font-size: 16px;
  padding: 15px 32px;
  display: inline-block;
  text-align: center;
  color: #ffffff;
  border-radius: 5px;
  float: right;
  margin-bottom: 20px; }

nav.icons div {
  margin-right: 15px; }

nav.icons i {
  margin-top: 5px; }

nav.icons div.search input[type='text'] {
  margin-right: 10px;
  border-radius: 6px;
  padding: 4px 12px;
  border-color: #eee;
  border-width: 1px; }

nav.icons div.search #search-button {
  margin-right: 10px; }

.card {
  float: left;
  margin-top: 10px;
  padding: 10px;
  width: 25%;
  height: 400px;
  background-color: #ffffff;
  border-color: #00aef9;
  border-width: 1px;
  box-shadow: #808e9b;
  margin-right: 10px; }

.card:hover {
  cursor: pointer;
  text-decoration: none; }

.card > img {
  width: 100%;
  height: 50%; }

.card .contentDescription {
  width: 100%;
  margin-top: 10px;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader {
  width: 100%;
  text-align: center;
  word-wrap: break-word; }

.card .contentDescription .articleCardHeader p {
  font-weight: bold;
  font-size: 20px;
  text-transform: capitalize; }

div.alert {
  width: 30%;
  height: 30px;
  display: inline-block;
  background-color: #bb321f;
  color: #ffffff;
  margin-left: 30%;
  margin-right: 30%;
  border-radius: 5px;
  margin-top: 10px;
  padding: 30px;
  font-size: 20px; }

.contents {
  background-color: #ffffff;
  margin-top: 0;
  min-height: 800px; }

.contents .contentTitle {
  width: 100%;
  text-align: center;
  padding: 10px; }

.contents .contentBody #articleBody {
  clear: both;
  float: left;
  margin-top: 15px;
  padding-left: 10%;
  padding-right: 10%; }

.contents .contentBody p {
  padding-left: 50px;
  padding-right: 50px;
  font-size: 20px;
  word-wrap: break-word;
  line-height: 1.5em; }

.contents .contentBody p img {
  width: 80%;
  margin-left: auto;
  margin-right: auto;
  display: block;
  height: 500px; }

.article-icons .icon {
  margin-right: 10px;
  display: inline-block; }

.contentDescription a {
  text-decoration: none; }

.contentDescription a:link {
  color: #2f3640; }

.contentDescription a:visited {
  color: #2f3640; }

.card .contentDescription .buttons {
  margin-bottom: 5px;
  position: absolute;
  margin-top: 10px; }

.card .contentDescription .buttons .redButton {
  background: #bb321f;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .contentDescription .buttons .lightBlueButton {
  background: #74b9ff;
  border: 0;
  color: #ffff;
  text-align: center;
  display: inline-block;
  font-size: 10px;
  margin: 5px;
  padding: 3px 12px; }

.card .articleDescription {
  height: 65px;
  margin-top: 5px; }

.container .allArticlecontents {
  background-color: #ffffff;
  margin-top: 0;
  height: 100%; }

.socialMedia {
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer; }
  .socialMedia .facebookButton {
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
    .socialMedia .facebookButton.facebook {
      color: #3b5998;
      border-color: #3b5998; }
      .socialMedia .facebookButton.facebook:hover {
        background-color: #3b5998;
        border-color: #3b5998;
        color: #fff; }
    .socialMedia .facebookButton.twitter {
      color: #1da1f2;
      border-color: #1da1f2; }
      .socialMedia .facebookButton.twitter:hover {
        background-color: #1da1f2;
        border-color: #1da1f2;
        color: #fff; }
    .socialMedia .facebookButton.googleplus {
      color: #dd4b39;
      border-color: #dd4b39; }
      .socialMedia .facebookButton.googleplus:hover {
        background-color: #dd4b39;
        border-color: #dd4b39;
        color: #fff; }
    .socialMedia .facebookButton.linkedin {
      color: #0077b5;
      border-color: #0077b5; }
      .socialMedia .facebookButton.linkedin:hover {
        background-color: #0077b5;
        border-color: #0077b5;
        color: #fff; }
    .socialMedia .facebookButton.whatsapp {
      color: #128c7e;
      border-color: #128c7e; }
      .socialMedia .facebookButton.whatsapp:hover {
        background-color: #128c7e;
        border-color: #128c7e;
        color: #fff; }
  .socialMedia .twitterButton {
    color: #1da1f2;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .twitterButton:hover {
    background-color: #1da1f2;
    color: #fff; }
  .socialMedia .linkedinButton {
    color: #0077b5;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .linkedinButton:hover {
    background-color: #0077b5;
    color: #fff; }
  .socialMedia .whatsappButton {
    color: #128c7e;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .whatsappButton:hover {
    color: #fff;
    background-color: #128c7e; }
  .socialMedia .emailButton {
    color: #dd4b39;
    padding: 2px;
    margin: 0.5rem 0.25rem;
    border-radius: 2px; }
  .socialMedia .emailButton:hover {
    color: #fff;
    background-color: #dd4b39; }

.edit-history-section {
  margin-top: 10px; }
  .edit-history-section .hideButton {
    background: #fff;
    padding: 3px;
    color: #000;
    border-radius: 3px;
    cursor: pointer;
    margin-bottom: 3px;
    font-size: 20px; }

.history-section {
  padding: 3px 15px 5px 0px;
  border-bottom: 1px solid #c5c5c5; }
  .history-section time {
    color: #9c9c9c;
    font-size: 12px;
    font-weight: bold; }

.emailButton:hover {
  color: #fff;
  background-color: #dd4b39; }

.likeButton {
  border: none;
  color: #9c9c9c;
  cursor: pointer;
  width: 15px;
  height: 15px; }

.likeButton:hover {
  color: #43c0f1; }

.buttonSpan {
  margin-right: 5px; }

.topImage input[type='file'] {
  cursor: inherit;
  display: block;
  min-height: 100%;
  min-width: 100%;
  opacity: 0;
  position: absolute;
  right: 0;
  text-align: right;
  top: 0;
  float: right; }

.topImage .file-upload {
  overflow: hidden;
  position: absolute;
  width: 100px;
  height: 100px; }

.topImage .file-upload .imageIcone {
  min-height: 100%;
  min-width: 100%;
  margin-left: 20%;
  color: #3b5998; }

.tags {
  padding: 10px;
  margin-left: 20px;
  background-color: #3b5998;
  color: #fff;
  border-radius: 5px;
  display: inline-block;
  cursor: pointer; }

#tags > input {
  border-radius: 5px;
  border: none;
  width: 178px;
  height: 36px;
  margin-bottom: 5px;
  padding: 10px;
  font-size: 15px;
  margin-right: 5px; }

#tags {
  width: 75%;
  float: left; }

#addTag {
  width: 10%;
  float: left; }

.tagList {
  padding: 2px;
  margin: 0.5rem 0.25rem;
  border-radius: 2px;
  background-color: #4b4b4b;
  color: #fff;
  display: inline-block;
  padding: 5px;
  cursor: pointer; }
