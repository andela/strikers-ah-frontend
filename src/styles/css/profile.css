@import url('https://fonts.googleapis.com/css?family=Open+Sans&display=swap');
*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Roboto Condensed', sans-serif;
}

body {
  background: #f0f0f0;
  font-family: 'Open Sans', sans-serif;
}

@keyframes showAddImage {
  from {
    background: transparent;
    opacity: 0;
  }
  to {
    background: #0788f1;
    opacity: 0.7;
  }
}

@keyframes modal {
  from {
    margin-top: 0;
    border-radius: 0;
    box-shadow: none;
  }
  to {
    margin-top: 3%;
    border-radius: 10px;
    box-shadow: 0px 0px 14px #333;
  }
}

.web-heading {
  color: #079bfd;
}

.profile-grid-off,
.profile-grid {
  padding: 2%;
  position: relative;
}
.profile-grid-off .user-account-info,
.profile-grid .user-account-info {
  padding: 1.5%;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
  background: white;
}
.profile-grid-off .user-account-info .statistics,
.profile-grid .user-account-info .statistics {
  width: 100%;
  border-top: 2px solid #ddd;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  text-align: center;
  margin-top: 1%;
  padding-top: 1%;
}
.profile-grid-off .user-account-info .statistics .stat-label,
.profile-grid .user-account-info .statistics .stat-label {
  text-decoration: none;
  color: #949292;
  display: inline-block;
}
.profile-grid-off .user-account-info .statistics .stat-label .value,
.profile-grid .user-account-info .statistics .stat-label .value {
  color: #333;
  font-weight: bold;
  font-size: 1.2em;
}
.profile-grid-off .user-account-info .user-account-flex,
.profile-grid .user-account-info .user-account-flex {
  position: relative;
  display: grid;
  grid-template-columns: 10em 1fr;
  grid-column-gap: 10px;
}
.profile-grid-off .user-account-info .user-account-flex .img-container,
.profile-grid .user-account-info .user-account-flex .img-container {
  position: relative;
  height: 10em;
  width: 10em;
  margin-right: 2em;
  text-align: center;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler {
  position: absolute;
  background: #ddd;
  bottom: 0;
  border: none;
  outline: none;
  left: 0;
  width: 100%;
  background: transparent;
  font-weight: bolder;
  color: transparent;
  font-size: 2em;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  .add-img-toggler:hover {
  animation-name: showAddImage;
  animation-duration: 0.5s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  color: white;
}
.profile-grid-off .user-account-info .user-account-flex .img-container img,
.profile-grid .user-account-info .user-account-flex .img-container img {
  width: 100%;
  height: 100%;
  cursor: pointer;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container
  img:hover
  ~ .add-img-toggler {
  z-index: 1;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label,
.profile-grid
  .user-account-info
  .user-account-flex
  .img-container:hover
  .profile-image-label {
  display: inline-block;
}
.profile-grid-off .user-account-info .user-account-flex .info-container,
.profile-grid .user-account-info .user-account-flex .info-container {
  padding: 1em 0;
  grid-area: 'info-container';
}
.profile-grid-off .user-account-info .user-account-flex .info-container .name,
.profile-grid .user-account-info .user-account-flex .info-container .name {
  font-weight: 500;
  font-size: 2em;
  text-transform: capitalize;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .username,
.profile-grid .user-account-info .user-account-flex .info-container .username {
  line-height: 40px;
  color: #949292;
}
.profile-grid-off .user-account-info .user-account-flex .info-container .bio,
.profile-grid .user-account-info .user-account-flex .info-container .bio {
  padding: 1% 10px 1% 0px;
  line-height: 30px;
  color: #535353;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow {
  float: right;
  margin-right: 10px;
  border: 1px solid #079bfd;
  width: 100px;
  padding: 5px;
  border-radius: 5px;
  color: #079bfd;
  margin-top: 10px;
  outline: none;
  font-size: 0.9em;
  transition: 0.2s;
}
.profile-grid-off
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover,
.profile-grid
  .user-account-info
  .user-account-flex
  .info-container
  .btn-follow:hover {
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  background: #079bfd;
  color: white;
  transform: translateY(-2px);
}
.profile-grid-off .upload-image-modal,
.profile-grid .upload-image-modal {
  position: absolute;
  width: 100%;
  height: 70vh;
  top: 0;
  left: 50%;
  background: rgba(86, 86, 87, 0.7);
  transform: translateX(-50%);
  z-index: 100;
}
.profile-grid-off .upload-image-modal .upload-page,
.profile-grid .upload-image-modal .upload-page {
  position: relative;
  width: 90%;
  margin-top: 3%;
  margin-left: auto;
  margin-right: auto;
  background: #eff2f5;
  min-height: 90%;
  box-shadow: 0px 0px 14px #333;
  border-radius: 10px;
  animation-name: modal;
  animation-duration: 0.3s;
}
.profile-grid-off .upload-image-modal .btn,
.profile-grid .upload-image-modal .btn {
  font-size: 18px;
  line-height: 18px;
  height: 30px;
  padding: 0 10px;
  border-radius: 1px;
  margin: 1%;
}
.profile-grid-off .upload-image-modal .btn-cancel,
.profile-grid .upload-image-modal .btn-cancel {
  margin-left: 10%;
  border: 1px solid brown;
  color: #f38b8b;
}
.profile-grid-off .upload-image-modal .btn-cancel:hover,
.profile-grid .upload-image-modal .btn-cancel:hover {
  background: brown;
  color: white;
}
.profile-grid-off .upload-image-modal .btn-upload,
.profile-grid .upload-image-modal .btn-upload {
  background: white;
  border: 1px solid #079bfd;
  color: #079bfd;
}
.profile-grid-off .upload-image-modal .btn-upload:hover,
.profile-grid .upload-image-modal .btn-upload:hover {
  background: #079bfd;
  color: white;
}
.profile-grid-off .upload-image-modal .dropzone,
.profile-grid .upload-image-modal .dropzone {
  width: 100%;
  background: #e1f0ff;
  height: 100px;
  margin-bottom: 1%;
  position: relative;
}
.profile-grid-off .upload-image-modal .dropzone p,
.profile-grid .upload-image-modal .dropzone p {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.profile-grid-off .upload-image-modal .crop .crop-section,
.profile-grid .upload-image-modal .crop .crop-section {
  width: 100%;
  display: grid;
  grid-template-columns: 60% 1% 30%;
  grid-column-gap: 1em;
}
.profile-grid-off .upload-image-modal .crop .cropper,
.profile-grid .upload-image-modal .crop .cropper {
  width: 100%;
  overflow: auto;
  height: 400px;
  text-align: center;
  border-radius: 10px;
}
.profile-grid-off .upload-image-modal .crop .separator,
.profile-grid .upload-image-modal .crop .separator {
  background: #dde4e9;
}
.profile-grid-off .upload-image-modal .crop .crop-preview,
.profile-grid .upload-image-modal .crop .crop-preview {
  width: 100%;
  overflow: hidden;
  text-align: center;
}
.profile-grid-off .upload-image-modal .crop .crop-preview canvas,
.profile-grid .upload-image-modal .crop .crop-preview canvas {
  border-radius: 10px;
  box-shadow: 0px 0px 14px #949292;
  border: 5px solid #ddd;
}
.profile-grid-off .left-profile,
.profile-grid .left-profile {
  background: white;
  box-shadow: 0px 0px 14px rgba(191, 191, 191, 0.24);
  border-radius: 10px;
}
.profile-grid-off .left-profile li,
.profile-grid .left-profile li {
  list-style-type: none;
  border-top: 1px solid #f0f0f0;
  line-height: 35px;
  padding-left: 10px;
}
.profile-grid-off .left-profile li .link,
.profile-grid .left-profile li .link {
  text-decoration: none;
  color: #949292;
}
.profile-grid-off .left-profile li .active,
.profile-grid .left-profile li .active {
  color: #079bfd;
}
.profile-grid-off .left-profile h3,
.profile-grid .left-profile h3 {
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  line-height: 40px;
  padding-left: 10px;
  background: rgba(9, 129, 241, 0.89);
  color: white;
}

.profile-grid {
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-column-gap: 2em;
  grid-template-areas: 'left-profile user-account-flex';
}

@media screen and (max-width: 1500px) {
  .profile-grid {
    width: 80%;
    margin: 2em auto;
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 10px;
    margin: 2em auto;
  }
}

@media screen and (max-width: 1000px) {
  .profile-grid {
    display: grid;
    grid-template-columns: 1fr 3fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
}

@media screen and (max-width: 815px) {
  .profile-grid {
    width: 80%;
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 4px;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid .left-profile {
    display: none;
  }
}

@media screen and (max-width: 500px) {
  .profile-grid-off {
    display: grid;
    grid-template-columns: 1fr;
    width: 98%;
    margin: 2em auto;
  }
  .profile-grid-off .user-account-flex {
    display: grid;
    grid-template-columns: 1fr;
    grid-column-gap: 0px;
    grid-template-areas: 'image-container' 'info-container';
  }
  .profile-grid-off .user-account-flex .info-container {
    display: block;
    width: 100%;
  }
  .profile-grid-off .user-account-flex .info-container .name {
    font-weight: 100;
    font-size: 1em;
  }
  .profile-grid {
    display: block;
  }
}

.text-center {
  text-align: center;
}
