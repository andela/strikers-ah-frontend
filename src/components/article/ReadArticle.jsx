/* eslint-disable react/destructuring-assignment */
import reactHtml from 'html-react-parser';
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import Header from './Header';
import '../../styles/css/article.css';
import { getOneArticle } from '../../redux/actions/articleAction';
import Author from './Author';
import SocialMedia from './SocialMedia';

/**
 *@author: Innocent Nkunzi
 * @returns {*} Articleform
 */
export class ReadArticle extends Component {
  /**
   * @author Innocent Nkunzi
   * @returns {*} componentDidmount
   */
  componentWillMount() {
    const { slug } = this.props.match.params;
    this.props.getOneArticle(slug);
  }

  /**
   * @author Innocent Nkunzi
   * @returns {*} component
   */
  render() {
    const singleArticle = this.props.article.article;
    if (singleArticle !== null && singleArticle !== undefined) {
      return (
        <Fragment>
          <Header />
          <div className="container">
            <div className="contents">
              <div className="contentTitle">
                {/* {article.title && <p>{reactHtmlparser(article.title)}</p>} */}
                <h1>{reactHtml(singleArticle.title)}</h1>
                <Author />
                <SocialMedia />
              </div>
              <div className="contentBody">
                <p>{reactHtml(singleArticle.body)}</p>
                <p id="styleImage">
                  <img src={singleArticle.image} alt="Article" />
                </p>
              </div>
            </div>
          </div>
        </Fragment>
      );
    }
    return (
      <div>
        <p>Single article</p>
      </div>
    );
  }
}
const mapStateToprops = state => ({
  article: state.Article,
});

export default connect(
  mapStateToprops,
  { getOneArticle },
)(ReadArticle);
