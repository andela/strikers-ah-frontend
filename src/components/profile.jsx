import React, { Component } from 'react';
import { connect } from 'react-redux';
// import LeftProfile from './common/leftProfile';
import RightProfile from './common/profileRightSide';
import { getUserProfile, getUserArticles } from '../redux/actions/userAction';
import '../styles/css/profile.css';
import { getLoggedInUser } from '../helpers/authentication';

/**
 * @author Mwibutsa FLoribert
 * @returns { * } --
 */
class Profile extends Component {
  /**
   * @author Mwibutsa Floribert
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {
      editProfile: { showForm: false, showImageForm: false },
      pagination: { pageSize: 5, currentPage: 0 },
    };
    const {
      getUserProfile: getProfile,
      getUserArticles: getArticles,
    } = this.props;
    const { username } = this.props;
    getProfile(username);
    getArticles(username);
  }

  /**
   * @author Mwibutsa Floribert
   * @param { * } nextPage --
   * @param { * } pages --
   * @param { * } prev --
   * @param { * } next --
   * @returns { * } --
   */
  paginate = (nextPage, pages, prev = false, next = false) => {
    const { pagination } = this.state;
    let activePage = pagination.currentPage;
    if (prev) {
      if (pagination.currentPage === 0) activePage = pages - 1;
      else activePage -= 1;
    } else if (next) {
      if (pagination.currentPage === pages - 1) activePage = 0;
      else activePage += 1;
    } else {
      activePage = nextPage;
    }
    this.setState({ pagination: { ...pagination, currentPage: activePage } });
  };

  toggleEditProfile = () => {
    const { editProfile } = this.state;
    this.setState({ editProfile: { showForm: !editProfile.showForm } });
  };

  toggleShowImage = () => {
    const { editProfile } = this.state;
    this.setState({
      editProfile: { showImageForm: !editProfile.showImageForm },
    });
  };
  /**
   * @author Mwibutsa FLoribert
   * @param { Integer } loggedInUserId --
   * @param { Integer } userProfileId --
   * @returns { Boolean } returns the comparison of logged in user id and user profile
   */

  isOwner = (loggedInUserId, userProfileId) => {
    return loggedInUserId === userProfileId;
  };

  /**
   * @author Mwibutsa FLoribert
   * @returns { JSX } --
   */
  render() {
    const { accountInfo: user } = this.props;
    const { articles } = this.props;
    let userArticles = [];
    if (Object.keys(articles).length) {
      userArticles = articles.articles;
    }
    let { profile } = user;
    profile = profile || user;
    const defaultUser = {
      name: 'Not Provided',
      image:
        'https://cdn.pixabay.com/photo/2013/07/13/10/44/man-157699__480.png',
      bio: '',
      followers: [],
      followings: [],
    };
    const isLoggedIn = !!getLoggedInUser();
    const owner = getLoggedInUser().id === profile.id;
    let ownerClass = 'profile-page';
    if (!isLoggedIn || !owner) {
      ownerClass += ' friend-profile';
    }
    const { editProfile, pagination } = this.state;
    return (
      <div className={ownerClass} test-data="profile-page" id="profile-page">
        <div className="profile-grid">
          {profile.email && (
            <RightProfile
              accountInfo={{
                ...defaultUser,
                ...profile,
                owner,
              }}
              editProfile={{
                showForm: editProfile.showForm,
                toggleEditProfile: this.toggleEditProfile,
                toggleShowImage: {
                  visible: editProfile.showImageForm,
                  toggleForm: this.toggleShowImage,
                },
              }}
              userArticles={userArticles}
              pagination={pagination}
              paginate={this.paginate}
            />
          )}
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    accountInfo: state.userProfile.user,
    articles: state.userArticles,
  };
};
export default connect(
  mapStateToProps,
  { getUserProfile, getUserArticles },
)(Profile);
