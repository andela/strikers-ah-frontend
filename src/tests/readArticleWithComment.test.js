/* eslint-disable no-dupe-keys */
import React from 'react';
import '../enzymeConfig';
import { ReadArticle } from '../components/article/ReadArticle';
import { getComponent, findByTestAttribute } from './profile/test-helpers';

describe('TEST READ ARTICLE WITH COMMENTS', () => {
  let wrapper;
  let instance;
  let props;
  beforeEach(() => {
    props = {
      comments: [{ id: 1, comment: 'this is good' }],
      match: { params: { slug: 'slug' } },
      getComments: jest.fn(),
      getOneArticle: jest.fn(),
      article: { article: [{ id: 1, slug: 'slug' }] },
      toggleEditCommentForm: jest.fn(),
      toggleEditHistory: jest.fn(),
      article: {
        article: {
          taglist: ['tag1', 'tag2'],
        },
      },
    };
    wrapper = getComponent(<ReadArticle {...props} />, {}, { ...props });
    instance = wrapper.instance();
    jest.spyOn(instance, 'toggleEditCommentForm');
    jest.spyOn(instance, 'toggleEditHistory');
  });
  afterEach(() => {
    instance.toggleEditCommentForm.mockClear();
    instance.toggleEditHistory.mockClear();
  });
  test('it should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to toggle edit comment form', () => {
    const elem = findByTestAttribute(wrapper, 'commentComponent');
    elem.props().toggleEditCommentForm(1, false);
    expect(instance.toggleEditCommentForm).toBeCalledTimes(1);
  });
  it('should be able to toggle edit edit history', () => {
    const elem = findByTestAttribute(wrapper, 'commentComponent');
    elem.props().toggleEditHistory(false);
    expect(instance.toggleEditHistory).toBeCalledTimes(1);
  });
  it('should be able to call edit history', () => {
    instance.toggleEditHistory(1, true);
  });
  it('should be able to add comments to the list when created', () => {
    instance.componentWillReceiveProps({ comment: 'this comment' });
  });
  it('should be able to add comments to the list when created1', () => {
    props.comments = [
      { id: 1, comment: 'this is good' },
      { id: 2, comment: 'this is good' },
      { id: 3, comment: 'this is good' },
    ];
    wrapper = getComponent(<ReadArticle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to add comments to the list when created1', () => {
    props.comments = [];
    wrapper = getComponent(<ReadArticle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should be able to add comments to the list when created1', () => {
    props.comments = false;
    wrapper = getComponent(<ReadArticle {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
