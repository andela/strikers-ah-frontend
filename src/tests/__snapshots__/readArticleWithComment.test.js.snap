// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST READ ARTICLE WITH COMMENTS it should render 1`] = `ShallowWrapper {}`;

exports[`TEST READ ARTICLE WITH COMMENTS should be able to add comments to the list when created1 1`] = `ShallowWrapper {}`;

exports[`TEST READ ARTICLE WITH COMMENTS should be able to add comments to the list when created1 2`] = `ShallowWrapper {}`;

exports[`TEST READ ARTICLE WITH COMMENTS should be able to add comments to the list when created1 3`] = `ShallowWrapper {}`;
